File: uri/uri_cpp/grafos/p2562.cpp
#include <iostream>
#include <sstream>
#include <string>
#include <stdlib.h>
#include <stdio.h>
#include <math.h>
#include <vector>

using namespace std;

int main() {
    int V, A, E, i, x, y, cont, topo;
    vector < vector <int> > G;
    vector <int> cor;
    vector <int> pilha;

    while (cin >> V) {
        cin >> A;

        for (i = 0; i < V; i++) {
            vector <int> row;
            G.push_back(row);
            cor.push_back(1);
        }
        for (i = 0; i < A; i++) {
            cin >> x;
            cin >> y;
            G[x-1].push_back(y-1);
            G[y-1].push_back(x-1);
        }
        cin >> E;
        E -= 1;
        pilha.push_back(E);
        cor[E] = 0;
        cont = 1;
        while (pilha.size() > 0) {
            topo = pilha.back();
            pilha.pop_back();
            for (i = 0; i < G[topo].size(); i++) {
                if (cor[G[topo][i]]) {
                    pilha.push_back(G[topo][i]);
                    cor[G[topo][i]] = 0;
                    cont += 1;
                }
            }
        }
        cout << cont << endl;
        G.clear();
        cor.clear();
        pilha.clear();
        cont = 0;
    }

    return 0;
}

File: uri/uri_python/grafos/p2562.py
while True:
	try:
		entry = input().split(' ')
		V = int(entry[0])
		A = int(entry[1])
		G = []
		cor = []
		for i in range(V):
			G.append([])
			cor.append(1)
		for i in range(A):
			entry = input().split(' ')
			G[int(entry[0])-1].append(int(entry[1])-1)
			G[int(entry[1])-1].append(int(entry[0])-1)
		E = int(input()) - 1
		pilha = [E]
		cor[E] = 0
		cont = 1
		while len(pilha) > 0:
			topo = pilha.pop()
			for i in range(len(G[topo])):
				if cor[G[topo][i]]:
					pilha.append(G[topo][i])
					cor[G[topo][i]] = 0
					cont += 1
			
		print(cont)
	except EOFError:
		break